websites/T/The Odin Project/presence.ts

```diff
@@ -0,0 +1,68 @@
+/* eslint-disable unicorn/prefer-dom-node-text-content */
+const presence = new Presence({
+		//The client ID of the Application created at https://discordapp.com/developers/applications
+		clientId: "1013489969379152022",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "https://i.imgur.com/gtBvHQH.png",
+			startTimestamp: browsingTimestamp,
+		},
+		lesson = document.querySelector(
+			"body > div.page-container.lesson > header > div > div > h1"
+		) as HTMLElement | null,
```

Comment:
Instead of using `HTMLElement` which is the base of all elements, you should use the specific element it is going to be. For example, in this case it will be `HTMLHeadingElement` as the last element in the selector is `h1`. Also, please use Type Generic for defining the type of a variable instead of using the `as` keyword - again, this is just something we prefer 
```suggestion
		lesson = document.querySelector<HTMLHeadingElement>(
			"body > div.page-container.lesson > header > div > div > h1"
		),
```

Reply:
Used this as a solution to innerText not working initially.

Reply:
Use `textContent` in this case

Reply:
i listened to all ur suggestion and tried implementing them and refactoring code have a look at my new commit. i also switched to textcontent